ibis/pyspark/compiler.py

```diff
@@ -0,0 +1,498 @@
+
+
+@compiles(ops.Exp)
+def compile_exp(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.exp(src_column)
+
+
+@compiles(ops.Sign)
+def compile_sign(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+
+    return F.when(src_column == 0, F.lit(0.0)) \
+        .otherwise(F.when(src_column > 0, F.lit(1.0)).otherwise(-1.0))
+
+
+@compiles(ops.Sqrt)
+def compile_sqrt(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.sqrt(src_column)
+
+
+@compiles(ops.Log)
+def compile_log(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.log(float(op.base.op().value), src_column)
+
+
+@compiles(ops.Ln)
+def compile_ln(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.log(src_column)
+
+
+@compiles(ops.Log2)
+def compile_log2(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.log2(src_column)
+
+
+@compiles(ops.Log10)
+def compile_log10(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.log10(src_column)
+
+
+@compiles(ops.Modulus)
+def compile_modulus(t, expr):
+    op = expr.op()
+
+    left = t.translate(op.left)
+    right = t.translate(op.right)
+    return left % right
+
+
+@compiles(ops.Negate)
+def compile_negate(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return -src_column
+
+
+@compiles(ops.Add)
+def compile_add(t, expr):
+    op = expr.op()
+
+    left = t.translate(op.left)
+    right = t.translate(op.right)
+    return left + right
+
+
+@compiles(ops.Divide)
+def compile_divide(t, expr):
+    op = expr.op()
+
+    left = t.translate(op.left)
+    right = t.translate(op.right)
+    return left / right
+
+
+@compiles(ops.FloorDivide)
+def compile_floor_divide(t, expr):
+    op = expr.op()
+
+    left = t.translate(op.left)
+    right = t.translate(op.right)
+    return F.floor(left / right)
+
+
+@compiles(ops.Power)
+def compile_power(t, expr):
+    op = expr.op()
+
+    left = t.translate(op.left)
+    right = t.translate(op.right)
+    return F.pow(left, right)
+
+
+@compiles(ops.IsNan)
+def compile_isnan(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.isnan(src_column)
+
+
+@compiles(ops.IsInf)
+def compile_isinf(t, expr):
+    import numpy as np
+    op = expr.op()
+
+    @pandas_udf('boolean', PandasUDFType.SCALAR)
+    def isinf(v):
+        return np.isinf(v)
+
+    src_column = t.translate(op.arg)
+    return isinf(src_column)
+
+
+@compiles(ops.ValueList)
+def compile_value_list(t, expr):
+    op = expr.op()
+    return [t.translate(col) for col in op.values]
+
+
+@compiles(ops.InnerJoin)
+def compile_inner_join(t, expr):
+    return compile_join(t, expr, 'inner')
+
+
+def compile_join(t, expr, how):
+    op = expr.op()
+
+    left_df = t.translate(op.left)
+    right_df = t.translate(op.right)
+    # TODO: Handle multiple predicates
```

Comment:
Does spark not accept `pred1 AND pred2 AND ... AND predN`-style predicates?

Reply:
It does accept it. I just haven't implemented it yet.

Reply:
I see.

Reply:
Can you write an xfailing test that will pass when this is implemented?